File: Backend/LibraryAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryAPI.Models
{
    public class Book
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Title is required")]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required(ErrorMessage = "Author is required")]
        [StringLength(100)]
        public string Author { get; set; } = string.Empty;

        [StringLength(1000)]
        public string? Description { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: Frontend/library-frontend/src/types/Book.ts
/**
 * Book interface representing a book entity
 */
export interface Book {
  id: number;
  title: string;
  author: string;
  description?: string;
  createdAt?: string;
  updatedAt?: string;
}

/**
 * Interface for creating a new book (without id)
 */
export interface CreateBookDto {
  title: string;
  author: string;
  description?: string;
}

/**
 * Interface for updating an existing book
 */
export interface UpdateBookDto {
  id: number;
  title: string;
  author: string;
  description?: string;
}
